### PR TITLE
Add an IPv6 classifier

### PR DESCRIPTION
**Pull request by ChristophWurst**
Fixes https://github.com/nextcloud/suspicious_login/issues/44

* [x] Add IPv6 to binary converter
* [x] Add random IPv6 generator
* [x] Store IP version with persisted model metadata
* [x] Add a dedicated IPv6 background job for training
* [x] Add dedicated training config presets for v4/v6
* [x] Load only the corresponding model on login
* [x] Admin setting overhaul

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Bildschirmfoto von 2019-09-09 16-06-53

**Comment by ChristophWurst** [comment]
@nicokaiser @Gatak @schmuuu @albgus hello :wave: 

Would anyone of you mind giving this a test run?

Here's a build: [suspicious_login.tar.gz](https://github.com/nextcloud/suspicious_login/files/3590929/suspicious_login.tar.gz)

Please back up your instance/db/app files before the upgrade 😬

Let me know about your results (either wait a bit for the background job or just fire off a `occ suspiciouslogin:train --v6`). Happy testing :v:

**Comment by Schmuuu** [comment]
Hi @ChristophWurst 

This app has NC17 as requisites and I receive:
> This app cannot be installed because the following dependencies are not fulfilled:
> 
>    Server version 17 or higher is required.

Can I safely change the info.xml and allow NC16.0.4 as well?
Can't test otherwise unfortunately. I only have one productive server.

**Comment by ChristophWurst** [comment]
Yes, correct. Nextcloud 17 is a hard requirement. Sorry for not mentioning this.

**Comment by ChristophWurst** [comment]
FWIW this branch still works well on my production instance without ipv6

Bildschirmfoto von 2019-09-10 08-58-34

File: lib/Db/LoginAddressAggregatedMapper.php [hunk 1]
```diff
@@ -109,6 +109,24 @@ public function findHistoricAndRecentIpv4(int $threshold, int $maxAge = 0): arra
...
 	}
 
+	public function hasSufficientIpV6Data(int $start) {
+		$qb = $this->db->getQueryBuilder();
+
+		$query = $qb
+			->select($qb->createFunction('COUNT(*)'))
+			->from($this->getTableName())
+			->where($qb->expr()->andX(
+				$qb->expr()->notLike('ip', $qb->createNamedParameter('_%._%._%._%')),
```
Comment (rullzer) [inline lib/Db/LoginAddressAggregatedMapper.php]: not for now. But would it not make more sense for a future PR to just set a flag in the table if it is v4 or v6?

Comment (ChristophWurst) [reply lib/Db/LoginAddressAggregatedMapper.php]: Yeah, that would speed up the queries a bit

**Comment by Gatak** [comment]
I'm still on NC 16.0.4. Was it possible to change the info.xml or should I wait until 17 is out? 

Thanks!

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/suspicious_login/pull/107#issuecomment-529647903 -> https://github.com/nextcloud/suspicious_login/pull/107#issuecomment-529792242

**Comment by Gatak** [comment]
Ok. You said both 'Year's and a hard requirement ... ;). I'll see if enough apps are updatable to 17 and maybe try the beta.

**Pull request merged**